libnavigation-core/src/test/java/com/mapbox/navigation/core/MapboxNavigationTest.kt

```diff
@@ -1898,4 +1900,32 @@ internal class MapboxNavigationTest : MapboxNavigationBaseTest() {
             routesPreviewController.previewNavigationRoutes(emptyList())
         }
     }
+
+    @Test
+    fun mapboxNavigationCreationSetsInstanceToHolder() {
+        mockkObject(LegacyMapboxNavigationInstanceHolder) {
+            createMapboxNavigation()
+            verify { LegacyMapboxNavigationInstanceHolder.onCreated(mapboxNavigation) }
+        }
+    }
+
+    @Test(expected = IllegalStateException::class)
+    fun mapboxNavigationIsNotCreatedIfHolderHasInstance() {
+        mockkObject(LegacyMapboxNavigationInstanceHolder) {
+            every { LegacyMapboxNavigationInstanceHolder.peek() } returns mockk(relaxed = true)
+            createMapboxNavigation()
+            verify(exactly = 0) {
+                LegacyMapboxNavigationInstanceHolder.onCreated(any())
+            }
+        }
```

Comment:
will it test the same?
```suggestion
     createMapboxNavigation()
     createMapboxNavigation()
```

Reply:
How? There is not verification that `onCreated` was not called the second time.